### ISSUE TITLE
Wrong decoding with JSON_BIGINT_AS_STRING

### ISSUE THREAD

**Issue by milo** [remicollet/pecl-json-c#6]
``` php
$json = '12345678901234567890';

var_dump(json_decode($json, FALSE, 512));
# Expected: float(...some value...)
# Result:   float(9.2233720368548E+18)

var_dump(json_decode($json, FALSE, 512, JSON_BIGINT_AS_STRING));
# Expected: string(20) "12345678901234567890"
# Result:   string(19) "9223372036854775807"
```

I'm using self compiled PHP 5.4.10 with `--disable-json` and current json-c master 0eab694a211fc84daf89f53d6ea1ca38f62b11ae on 32bit Debian stable.

**Comment by Swader** [comment]
This is ridiculous, 2 years in and BIGINT to string implementation still isn't properly added? This is **not** a drop in replacement if there is no feature parity.

> I mean no encoder will ever generate such data.

Well, data that exceeds these bounds WAS generated by the company whose API I'm using, and its not up to you to have such opinions and avoid features based on your instincts when developing a "drop in" replacement that's so widespread. Please, either implement this properly, it's been 2 years since this was reported, or pull your extension and make clear in the docs that it is not to be used as a drop in replacement unless you're prepared to deal with notice catching and ignoring data.

**Comment by remicollet** [comment]
2 years waiting for your patch...
Still don't see any...
Yes this is ridiculous.

**Comment by Swader** [comment]
My patch? Look, if I knew how to do it I would, instantly, but the fact remains that you are claiming "drop in replacement" while NOT having feature parity. That is, by definition, not a drop in replacement. Considering a lot of linux distros now bundle this extension instead of the official ones, breakage IS possible (as I have witnessed on my own work, and others above), not to mention that this makes the official PHP docs incorrect.

I don't understand the reason for the existence of this extension, or the reason for why it doesn't have the same functionality as the open source extension it claims to replace, but it's a simple fact that his *must not* be advertised as a drop in replacement because that's not what it is.

I can't code C, but if I can help in any possible way to get this ball rolling, please let me know, I'll do anything in my power. Issues like these is why people make fun of PHP, a disparity such as this one is worse than the needle/haystack nonsense so let's get it out of the way, yes?

**Comment by remicollet** [comment]
All behavior changes are explained on https://github.com/remicollet/pecl-json-c/blob/master/README.md

> I don't understand the reason for the existence of this extension

https://bugs.php.net/63520

Problem is now solved in PHP 7. (with another new parser, from "jsond")

**Comment by Swader** [comment]
I didn't know about jsond. With that in mind - do you have plans to ever fix this for versions <7, or will this be abandoned?

I would obviously prefer it still be fixed - otherwise the minimum PHP requirement for libraries parsing numerals over 64 bits will have to be PHP 7, which is ridiculous to have due to a single bugged function.

**Comment by oerdnj** [comment]
Whoa @Swader, you can't code C, you can't spend few minutes searching for a reasoning behind this (if you type "php5 json" in the google, it's 6th link), you can't read a pecl-json-c's README file, and yet the thing you are really good at is behaving like a rude person. A very nice entré of you.

**Comment by Swader** [comment]
I have now read all the relevant materials in depth. I apologize for barging in without having read the relevant materials first.

I am, however, still unclear on why this functionality supporting >64bit ints wasn't simply taken from the original software which, if I'm not mistaken, has publicly available source code? Can you ELI5 why not reuse the implementation seen there?

**Comment by rcousens** [comment]
We also have hit this issue moving from 5.4 to 5.5. 

Frustrating to say the least but only because it broke our expectations of how large integers were handled.

**Comment by remicollet** [comment]
FYI see 3d0415c88cc2a865af54eaf6a613c89492c7b8be
